pyproject.toml

```diff
@@ -304,13 +304,17 @@ exclude= [
         ]
 
 [tool.ruff]
-lint.select = ["ALL"]
 exclude=[
     "astropy/extern/*",
     "*_parsetab.py",
     "*_lextab.py"
 ]
-lint.ignore = [  # NOTE: non-permanent exclusions should be added to `.ruff.toml` instead.
+
+[tool.ruff.lint]
+preview = true
+explicit-preview-rules = true
+select = ["ALL"]
+ignore = [  # NOTE: non-permanent exclusions should be added to `.ruff.toml` instead.
```

Comment:
You should also open a pull request to move these settings from the `[tool.ruff]` section to the `[tool.ruff.lint]` section.